### PR TITLE
core[minor]: Add utility code to create tool examples

### PR DESCRIPTION
**Pull request by hwchase17**

### CODE REVIEW

File: libs/core/langchain_core/utils/tool_examples.py [hunk 1]
```diff
@@ -0,0 +1,111 @@
+import uuid
```
Comment (eyurtsev) [inline libs/core/langchain_core/utils/tool_examples.py:1]: Look good. Do we think it's best to create tool_examples or put into function_calling module?

Comment (hwchase17) [reply libs/core/langchain_core/utils/tool_examples.py]: ya moved

File: libs/core/langchain_core/utils/function_calling.py [hunk 2]
```diff
@@ -311,3 +318,104 @@ def convert_to_openai_tool(
...
+    """A representation of an example consisting of text input and expected tool calls.
+
+    For extraction, the tool calls are represented as instances of pydantic model.
+    """
+
+    input: str  # This is the example text
+    tool_calls: List[BaseModel]  # Instances of pydantic model that should be extracted
+
+
+def tool_example_to_messages(example: Example) -> List[BaseMessage]:
```
Comment (baskaryan) [inline libs/core/langchain_core/utils/function_calling.py:333]: does input need to be dict, why not just have `input` and `tool_calls` as args

Comment (hwchase17) [reply libs/core/langchain_core/utils/function_calling.py:333]: idk @eyurtsev any thoughts?

Comment (eyurtsev) [reply libs/core/langchain_core/utils/function_calling.py:351]: yeah this can be unpacked

**Pull request merged**